logging/testing/util.go

```diff
@@ -27,13 +27,17 @@ import (
 )
 
 // TestLogger gets a logger to use in unit and end to end tests
-func TestLogger(t *testing.T) *zap.SugaredLogger {
+func TestLogger(i interface{}) *zap.SugaredLogger {
 	opts := zaptest.WrapOptions(
 		zap.AddCaller(),
 		zap.Development(),
 	)
 
-	return zaptest.NewLogger(t, opts).Sugar()
+	switch v := i.(type) {
+	case *testing.B:
+		return zaptest.NewLogger(v, opts).Sugar()
+	}
+	return zaptest.NewLogger(&testing.T{}, opts).Sugar()
```

Comment:
this seems wrong, shouldn't it be `i.(*testing.T)`?

Reply:
`i.(*testing.T)` and `&testing.T{}` both works

i dint use `i.(*testing.T)` because again need to add condition check in order to make sure its of type `testing.T` 
```
v, ok := i.(*testing.T)
```
and here as we know TestLogger can get type `testing.T` or `testing.B` so used directly

Reply:
but you're replacing the actual testing.T object with a new one. The old one had its state.

Reply:
Ah okay modified thank you